### ISSUE TITLE
Bitcode in the xcode 7.0.1

### ISSUE THREAD

**Issue by arasu01** [ParsePlatform/ParseUI-iOS#169]
When i am trying to archive the project for uploading to App Store. The following error occurs

ld: bitcode bundle could not be generated because '/Users/Parse/ParseUI.framework/ParseUI(PFImageView.o)' was built without full bitcode. All object files and libraries for bitcode must be generated from Xcode Archive or Install build for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by nlutsenko** [comment]
This sounds strange.
Any chance you can try using ParseUI via Cocoapods or building it yourself from source?

**Comment by nlutsenko** [comment]
Another person confirmed this, which looks like it does indeed is an issue.
It does indeed look like we are missing bitcode slice for the precompiled binary of ParseUI.framework.

Will fix it, but the current workaround is to use Cocoapods, which builds from source.

**Comment by pocketdeveloper** [comment]
I just checked the ParseUI.framework information, apparently the version is 1.1.6, but I downloaded the latest SDK.

**Comment by nlutsenko** [comment]
Yup, the current latest release is 1.1.6, milestone attached here means that we are aiming to fix this in a specific version (1.1.7 at this point).

**Comment by pocketdeveloper** [comment]
Just added Cocoapods to my project, deleted every Parse framework, used pod 'Parse', but I only got Parse and Bolts folders. ParseUI and ParseCrashReporting where not installed.

**Comment by nlutsenko** [comment]
This should help. All of these are separate frameworks, so they have dedicated pods. :grin:
```ruby
pod 'ParseUI'
pod 'ParseCrashReporting'
```

**Comment by pocketdeveloper** [comment]
@nlutsenko you rock! 🙌🙌🙌🙌 Thanks!

**Comment by nlutsenko** [comment]
:+1: Glad it helped, will still keep this issue open as we indeed should embed bitcode in the binary.

**Comment by CSCampbell** [comment]
I've got the same issue and have just downloaded the latest version which is 1.1.6. When might the fix be available?

**Comment by StephenWilson** [comment]
Can somebody at Parse please recompile the project and replace the bitcode incompatible file publicly available for download at https://www.parse.com/downloads/ios/parse-library/latest asap.

**Comment by CSCampbell** [comment]
Hello Stephen,

Thank you for your action. It is much appreciated. I can see that you are providing an excellent service, thank you very much.

In solving the problem. I followed the indication that using cocoapods might work. I didn't know what cocoapods was, however, I investigated and two hours later I managed to solve the problem for this app. However, it did cross my mind that all my other apps, would suffer the same problem until the new code fix arrived. Also, I imagine there will be very many others who will suffer the same fate.

Thank you for prompting them.

Best regards,

 Colin Campbell
cscampbell@btinternet.com

CONFIDENTIALITY: This message is private and confidential. If you have received this message in error, please notify us and remove it from your system.

**Comment by xissburg** [comment]
I also have this issue with ParseUI 1.1.6

**Comment by renman2112** [comment]
Any word on getting this fixed guys???

**Comment by aldmitriev** [comment]
>>I also have this issue with ParseUI 1.1.6. It works on Xcode 7.0 and fails in 7.0.1 and 7.1. I am not using Cocoapods, just dragging the framework into the project.

Hello, guys.
I have same problem with  Parse UI 1.9.1

ld: bitcode bundle could not be generated because '/%MyFolders%/Frameworks/Parse/ParseUI.framework/ParseUI(PFCollectionViewCell.o)' was built without full bitcode. All object files and libraries for bitcode must be generated from Xcode Archive or Install build for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)

I downloaded framework from Parse.com. just dragging the framework into the project (I don't use Cocoapods)
It compiled good (Build command), but I can't Archive that.

**Comment by Murf** [comment]
Same issue with 1.9.1, any ETA on a fix please?
An no I dont want to build from source or use Cocoa pods thanks anyway.

**Comment by waruss321** [comment]
To solve the problem you can just do the following. 

Build Settings >
Build Options >
Enable Bit Code = NO

:)

**Comment by aldmitriev** [comment]
Is it solution or fast fix?
Is it good to switch off bit code for WatchOS 2 app?

**Comment by nlutsenko** [comment]
watchOS 2 app - probably not, but you can't quite use ParseUI on watchOS, so everything should be fine. You can safely disable bitcode for the main app.
The other 2 workarounds:
- use CocoaPods
- build from source (which would also mean that you need to use CocoaPods, but to build the original framework only)

We are aiming to release the update late this week/early next week that will include the fix for this issue.

**Comment by aldmitriev** [comment]
Thank you everyone. Thank you Nikita. Let's try CocoaPods (
Please tell me more about it.
I don't use Parse UI in WatchOS2 app, but I can't just exclude the ParseUI. So I can build it from a source code only. And I need to use CocoaPods for that. And I can switch ON bitcode in this case. Do I understand right?
Or need I to disable bitcode anyway?
Thanks

**Comment by nlutsenko** [comment]
You are correct, if you use CocoaPods - there is no need to disable bitcode.

**Comment by aldmitriev** [comment]
@nlutsenko Thank you so much. I'll try

**Comment by aldmitriev** [comment]
@nlutsenko Hello again, Nikita. Sorry for stupid question, but it's so annoying.
 I tried to build it with Cocoapods. But I have some troubles with build. Could you suggest Pods config for Share extension and WatchOS app? Maybe some tricks? Maybe best practices to use Parse SDK with cocoapods?
Look at issue screenshot in attachment.
I don't understand 'Why'. Thank you!
I use 
```
target 'MyAPP' do
pod 'Parse'
pod 'Bolts'
pod 'ParseFacebookUtilsV4'
pod 'ParseTwitterUtils'
pod 'ParseUI'
pod 'ParseCrashReporting'
end

target 'MyAPPTests' do
end

target 'MyAPPToday' do
end

target 'MyAPPShare' do
pod 'Parse'
pod 'Bolts'
pod 'ParseFacebookUtilsV4'
pod 'ParseTwitterUtils'
pod 'ParseUI'
pod 'ParseCrashReporting'
end

target 'MyAPPWatch' do

end

target 'MyAPPWatch Extension' do
#platform :watchos, '2.0'
pod 'Parse'
pod 'Bolts'
pod 'ParseFacebookUtilsV4'
pod 'ParseTwitterUtils'
pod 'ParseUI'
pod 'ParseCrashReporting'
end
```
![screen](https://cloud.githubusercontent.com/assets/1948486/10954595/8b34d2b2-835f-11e5-9a7c-20063c2028f1.png)

Thank you!

**Comment by innopage** [comment]
Any update about this? I don't use cocoapods, but am eager to enable BitCode support in my next app update.

**Comment by nlutsenko** [comment]
Hey guys, ParseUI 1.1.7 just got released, it includes the fix for this issue - please download the ParseUI.framework from the releases page on this repo.

**Comment by zsll** [comment]
@nlutsenko Still having the same error with 1.1.7. Can you please confirm?

**Comment by nlutsenko** [comment]
Doing the second verification... I am 95% confident that it should be fixed at this point.

**Comment by nlutsenko** [comment]
@zsll Can you please make sure that you are downloading from releases on this repo and not from parse.com? I just retested this with an empty project and it works great!

**Comment by zsll** [comment]
Yes. I restarted my Xcode and it works this time. Thank you @nlutsenko.

**Comment by vTrip** [comment]
rookie question i know, but how do i download the latest release from the repo. not really familiar with downloading from repos. @nlutsenko

**Comment by aldmitriev** [comment]
@vTrip Look at releases page https://github.com/ParsePlatform/ParseUI-iOS/releases
It's only ParseUI.
Also you need to download all your needed parts separately (ParseSDK, Bolts etc)

**Comment by vTrip** [comment]
i read above and was wondering why it matters that we need to download from the repo and not from parse.com 
is this because the builds have not yet been approved therefore not publish to parse.com

**Comment by nlutsenko** [comment]
TL;DR Recommended way of using our SDKs right now is via CocoaPods or from source.

Here is the long list explaining our(my) thought process and what is the reasoning behind this.
It all is simple and is based on the following:
- At this point we have 5 frameworks forming the Parse SDK for iOS/OS X. They all live in separate repositories to give us an ability to have different release cycles.
- As you can still see - https://parse.com/docs/downloads page doesn't yet support different versioning for different frameworks, and it uses the version of the main framework currently.
- Having different downloads for the same version of the framework is quite confusing, since you don't even notice that there was an update (updating ParseUI framework in that download would use this case).

If we combine all of this - we get to the point, where the source of truth and the latest version is guaranteed to be on GitHub, since every single piece of all our SDKs is open source. And, zip file with the latest version of the main framework and not guaranteed to be latest versions of other frameworks is available as a download on Parse.com.

> Builds not approved...
There is not really an approval process. :grin:

**Comment by aldmitriev** [comment]
@nlutsenko Thanks, Nikita! Could you share example of Podfile? and optional step-by-step migration process?
I tried to use Cocoapods for Watchos2 and still have problems.
Need I any additional steps?
Is it possible to use it with watchos2?
I tried to add watchos platform, but got a message about Watchos 2.0
So I see that pods tring to use Bolts as a iOS SDK, not Watchos. And I can't fix it manually
Thank you!
```
 target 'MainApp' do
  pod 'Parse'
  pod 'ParseUI'
  pod 'ParseFacebookUtilsV4'
  pod 'ParseTwitterUtils'
end

target 'MainAppTests' do

end

target 'MainAppToday' do

end

target 'MainAppShare' do
  pod 'Parse'
  pod 'ParseUI'
  pod 'ParseFacebookUtilsV4'
  pod 'ParseTwitterUtils'
end

target 'MainAppWatch' do

end

target 'MainAppWatch Extension', :exclusive => true do
#  platform :watchos
# I tried to add watchos platform, but got a message about Watchos 2.0
# So I see that pods tring to use Bolts as a iOS SDK, not Watchos. And I can't fix it manually
  pod 'Parse'
  pod 'ParseUI'
  pod 'ParseFacebookUtilsV4'
  pod 'ParseTwitterUtils'
end
```

**Comment by nlutsenko** [comment]
Put out some information in ParsePlatform/Parse-SDK-iOS-OSX#179

**Comment by vTrip** [comment]
"ld: bitcode bundle could not be generated because '/Users/vincetripodi/Developer/TravelAppTest (original)/ParseUI.framework/ParseUI(PFLogInViewController.o)' was built without full bitcode. All object files and libraries for bitcode must be generated from Xcode Archive or Install build for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation) "


My version for ParseUI is 1.1.6 however the Parse.framework and Bolts.framework are the latest 1.10.1

Will these steps add the latest build to my project ?

ParseUI can also be built from source and supports Xcode subproject referencing.
Follow these steps to build and run via source code:

Download the source code via git clone or in an archive
Run pod install in the repo root to download all the dependencies
Open ParseUI.xcworkspace
Build and Run ParseUIDemo target



Regards,

Vince Tripodi

**Comment by nlutsenko** [comment]
The problem is fixed in ParseUI 1.1.7, so you need you to update.
The latest SDK download from https://parse.com/docs/downloads has all the latest frameworks at this point.

**Comment by vTrip** [comment]
I have achieved this now. Thank you for the assistance.


Regards,

Vince Tripodi
